### PR TITLE
Remove deprecated ArithmeticOperation

### PR DESCRIPTION
**Pull request by daxfohl**
@MichaelBroughton

### CODE REVIEW

**Comment by daxfohl** [comment]
Oof, that looks nontrivial (at least for me -- I'd need to ramp up on how to work with ipynb files, which I don't have time for right now). Anybody who wants to fork from here, I know there's a shor.py that probably had the same change when I originally deprecated `ArithmeticOperation`: https://github.com/quantumlib/Cirq/pull/4702/files#diff-9d5e940cf487ce1f28d94c8a32b7c4c2a5da812e72b48aa6e5db733e3f511066

**Comment by daxfohl** [comment]
@MichaelBroughton I figured out how to do this. However I'm getting an error saying `cirq.ArithmeticGate` is not defined. But it is. And it works locally. Any idea?

**Comment by pavoljuhas** [comment]
> However I'm getting an error saying cirq.ArithmeticGate is not defined. But it is. And it works locally. Any idea?

I am new to the notebook test machinery, but the test may be running with a released cirq.
Perhaps the shor.ipynb notebook needs to be added to NOTEBOOKS_DEPENDING_ON_UNRELEASED_FEATURES -

https://github.com/quantumlib/Cirq/blob/df99587d648e70206e5860efdbbe73a4182062f7/dev_tools/notebooks/isolated_notebook_test.py#L43-L59

**Comment by daxfohl** [comment]
@MichaelBroughton @pavoljuhas It turns out this is not ready to be removed. https://github.com/quantumlib/Cirq/pull/4702 was started in November but did not get merged until May, which was after 14.1 went out. These deprecations need to be bumped to 0.16 instead.

**Comment by daxfohl** [comment]
v0.15 is out, so reopening.

File: docs/experiments/shor.ipynb [hunk 1]
```diff
@@ -406,11 +414,14 @@
...
+    "# Define an adder gate for two 2D input and target qubits.\n",
+    "adder = Adder(input_register=[2, 2], target_register=[2, 2])\n",
+    "\n",
     "# Define the circuit.\n",
     "circ = cirq.Circuit(\n",
-    "    cirq.X.on(qreg1[0]),\n",
-    "    cirq.X.on(qreg2[1]),\n",
-    "    Adder(input_register=qreg1, target_register=qreg2),\n",
+    "    cirq.ops.X.on(qreg1[0]),\n",
+    "    cirq.ops.X.on(qreg2[1]),\n",
```
Comment (dstrain115) [inline docs/experiments/shor.ipynb:423]: Revert the addition of the ops module.

**Review by dstrain115 — APPROVED** [review]

**Pull request merged**